src/core/defcomp.jl

```diff
@@ -53,8 +57,9 @@ function _generate_run_func(module_name, comp_name, args, body)
     (p, v, d, t) = args
 
     if @capture(t, tname_::ttype_)
-        if ttype == :Timestep
-            ttype = :(Mimi.Timestep)
+        mimi_type = get(timestep_type_dict, ttype, nothing)
```

Comment:
The variable name is capitalized (Timestep_Type_Dict) but used as lower case here, indicating that this code has not been tested...